src/frontend/src/lib/stores/authentication.store.ts

```diff
@@ -57,6 +62,20 @@ export const authenticationStore: AuthenticationStore = {
         throw new Error("Not initialized");
       }
       initialized.agent.replaceIdentity(authenticated.identity);
+      // Create the OpenID nonce using the identity's principal
+      void createAnonymousNonce(authenticated.identity.getPrincipal()).then(
```

Comment:
At the beginning I didn't want to make this async, but the more I think about it, the more I'm leaning towards making `set` async and setting salt and nonce as non-optional in the `Authenticated` type.

Wdyt @sea-snake ?